src/main/java/com/aws/greengrass/shadowmanager/ShadowManagerDAO.java

```diff
@@ -11,8 +11,12 @@
 
 import java.util.List;
 import java.util.Optional;
+import java.util.concurrent.Semaphore;
 
 public interface ShadowManagerDAO {
+    int maxPermits = 10;
```

Comment:
nit: might need to increase this - we have some configuration for number of operations per second. May want to just use that (even if that includes read operations)

Reply:
Should not this be the number of threads operating on DB (for write) at any given time? Currently, we have three threads that can do DB write (IPC thread, Sync and service thread), so I had the max limit as 10.

Reply:
We might want to add a comment as to why that number is 10. I am pretty sure someone would see this in the future and don't know why it is 10 rather than 2 or 100. Is there a way to dynamically set the value based on the number of threads being used to access the DAO?

What if the number of threads gets increased to something higher than 10?

Reply:
right - for some reason I thought there would be a thread per request from IPC